contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -678,4 +678,25 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
         
         return volumeGUIDPath.substring(VOLUME_GUID_PATH_PREFIX.length(), volumeGUIDPath.length() - VOLUME_GUID_PATH_SUFFIX.length());
     }
+    
+    /**
+     * 
+     * This function retrieves the full path of the executable file of a given process.
+     * 
+     * @param hProcess
+     * 			Handle for the running process
+     * @param dwFlags
+     * 			0 - The name should use the Win32 path format.
+     * 			1(PROCESS_NAME_NATIVE) - The name should use the native system path format. 
+     * 
+     * @return the full path of the process's executable file of null if failed. To get extended error information, 
+     * 		   call GetLastError. 
+     */
+    public static final String QueryFullProcessImageName(HANDLE hProcess, int dwFlags) {
+        char[] path = new char[WinDef.MAX_PATH];
+        IntByReference lpdwSize = new IntByReference(path.length);
+        if (Kernel32.INSTANCE.QueryFullProcessImageName(hProcess, 0, path, lpdwSize))
+            return new String(path).substring(0, lpdwSize.getValue());
+    	return null;
```

Comment:
This is incorrect. This should be throwing an error if the `QueryFullProcessImageName` call fails.